_posts/2020-05-30-Metal3-Cluster-Deployment.md

```diff
@@ -0,0 +1,544 @@
+kubectl get metal3cluster ${CLUSTER_NAME} -n metal3
+```
+This will return the cluster deployed, and you can check the cluster details by describing the returned resource.
+
+Here is what a `Cluster` resource looks like :
+```console
+kubectl describe Cluster ${CLUSTER_NAME} -n metal3
+```
+```yaml
+Name:         eko
+Namespace:    metal3
+Labels:       <none>
+Annotations:  <none>
+API Version:  cluster.x-k8s.io/v1alpha3
+Kind:         Cluster
+Metadata:
+  Creation Timestamp:  2020-05-22T18:19:54Z
+  Finalizers:
+    cluster.cluster.x-k8s.io
+  Generation:  2
+  Managed Fields:
+    API Version:  cluster.x-k8s.io/v1alpha3
+    Fields Type:  FieldsV1
+    fieldsV1:
+      f:spec:
+        .:
+        f:clusterNetwork:
+          .:
+          f:pods:
+            .:
+            f:cidrBlocks:
+          f:services:
+            .:
+            f:cidrBlocks:
+        f:controlPlaneRef:
+          .:
+          f:apiVersion:
+          f:kind:
+          f:name:
+        f:infrastructureRef:
+          .:
+          f:apiVersion:
+          f:kind:
+          f:name:
+    Manager:      OpenAPI-Generator
+    Operation:    Update
+    Time:         2020-05-22T18:19:54Z
+    API Version:  cluster.x-k8s.io/v1alpha3
+    Fields Type:  FieldsV1
+    fieldsV1:
+      f:metadata:
+        f:finalizers:
+          .:
+          v:"cluster.cluster.x-k8s.io":
+      f:spec:
+        f:controlPlaneEndpoint:
+          f:host:
+          f:port:
+      f:status:
+        .:
+        f:infrastructureReady:
+        f:phase:
+    Manager:         manager
+    Operation:       Update
+    Time:            2020-05-22T18:19:54Z
+  Resource Version:  5795164
+  Self Link:         /apis/cluster.x-k8s.io/v1alpha3/namespaces/metal3/clusters/eko
+  UID:               b1e342ee-7f53-4dca-8959-8665aa74a26a
+Spec:
+  Cluster Network:
+    Pods:
+      Cidr Blocks:
+        192.168.0.0/18
+    Services:
+      Cidr Blocks:
+        10.96.0.0/12
+  Control Plane Endpoint:
+    Host:  192.168.111.249
+    Port:  6443
+  Control Plane Ref:
+    API Version:  controlplane.cluster.x-k8s.io/v1alpha3
+    Kind:         KubeadmControlPlane
+    Name:         eko
+    Namespace:    metal3
+  Infrastructure Ref:
+    API Version:  infrastructure.cluster.x-k8s.io/v1alpha4
+    Kind:         Metal3Cluster
+    Name:         eko
+    Namespace:    metal3
+Status:
+  Infrastructure Ready:  true
+  Phase:                 Provisioned
+Events:                  <none>
+```
+
+### Provision Controlplane
+
+This script, located at the path - ${metal3-dev-env}/scripts/v1alphaX/provision_clusters.sh, provisions the controlplane or the master of the cluster. As part of the controlplane provisioning, the generated manifests are for cluster, controlplane and machinedeployment ( creates a MachineSet, which creates a Machine that manages the master node ). It creates a `KubeadmControlPlane` resource and a `Metal3MachineTemplate` resource. To check if you have a successful deployment, try this - 
```

Comment:
Path with `back ticks`.

"Control plane" are 2 words, I don't think they should go together except in the references to resources.

More importantly, I think we should elaborate a bit more on what all these resources are, what's their purpose, why are they created.